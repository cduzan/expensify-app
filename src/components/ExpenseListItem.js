// Third Party Dependencies
import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
        <div>
            <Link to={`/edit/${id}`}>
                <h3>{description}</h3>
            </Link>
            <p>{amount} - {createdAt}</p>
        </div>
);

// Note: connect() without mapStateToProps doesn't pass any redux state but it allows you to use dispatch in the props
export default ExpenseListItem;
