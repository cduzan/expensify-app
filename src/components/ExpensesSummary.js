import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

// Note: props can now access what is passed from the redux state in the mapStateToProps function
export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00');
    return (
        <div>
            <h1>
                Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}
            </h1>
        </div>
    )
};

// Note: standard practice for the function that maps the redux state to the component props
const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

// Note: first function is for the redux state mapping
//       second function is for the component you're sending to the Higher Order Compnent
export default connect(mapStateToProps)(ExpensesSummary);