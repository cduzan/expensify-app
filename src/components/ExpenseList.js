import React from 'react';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

// Note: props can now access what is passed from the redux state in the mapStateToProps function
export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map((expense) => (
                    <ExpenseListItem key={expense.id} {...expense} />
                ))
            )
        }
    </div>
);

// Note: standard practice for the function that maps the redux state to the component props
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

// Note: first function is for the redux state mapping
//       second function is for the component you're sending to the Higher Order Compnent
export default connect(mapStateToProps)(ExpenseList);