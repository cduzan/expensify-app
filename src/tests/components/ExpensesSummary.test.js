import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary with 0 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={0} expensesTotal={0} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={1405} />);
    expect(wrapper).toMatchSnapshot();
});