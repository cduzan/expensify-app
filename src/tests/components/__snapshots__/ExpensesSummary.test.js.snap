// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 0 expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $14.05
  </h1>
</div>
`;
